js/src/components/EllipseGraphNode.vue

```diff
@@ -87,14 +94,45 @@ export default class EllipseGraphNode extends Vue {
 
   /** The size of the node, in terms of it's radius along the x, y axis. */
   get size() {
-    // Arbitrarily chosen magic constants to make things look good
-    this.rx = Math.min(Math.max(this.computedTotalWidth, 25), 50);
-    this.ry = Math.min(Math.max(this.computedTotalHeight - 12, 20), 35);
-    const bounds = { rx: this.rx, ry: this.ry };
+    var bounds = {rx: 0, ry: 0};
+
+    if(!this.hover){
+      // Arbitrarily chosen magic constants to make things look good
+      bounds = {
+        rx: Math.min(Math.max(this.computedTotalWidth, 25), 50),
+        ry: Math.min(Math.max(this.computedTotalHeight - 12, 20), 35)
+      };
+    } else {
+      // custom set by user visualizer (i.e. CSPVisualizer.vue)
+      bounds = {
+        rx: this.computedTotalWidth,
+        ry: this.computedTotalHeight
+      };
+    }
+
+    // this file was originally ellipse but now changed to rectangle with semi circle for its sides, so times the previous
+    // radius by 2 to make it width for the new rectangle
+    bounds.rx *= 2;
```

Comment:
I think it is slightly misleading that the EllipseGraphNode.vue class actually creates rectangles with rounded corners instead of ellipses. Maybe it is best to create a new RoundedRectangleGraphNode class which uses rectangle width and height directly.